### PR TITLE
[11.0][IMP] Add hook to hr_attendance_autoclose

### PR DESCRIPTION
**Pull request by jarroyomorales**
This PR adds a hook when autoclosing attendances.  This allows other modules to perform some other action when an autoclose happens.
Also adds a boolean field to disable autoclosing for certain employees.

### CODE REVIEW

**Comment by jarroyomorales** [comment]
@aheficent What do you think?

**Review by AaronHForgeFlow — COMMENTED** [review]
@jarroyomorales  :+1:  to this, as long this keep the original functionality and allows to extend it.

File: hr_attendance_autoclose/models/hr_attendance.py [hunk 1]
```diff
@@ -39,18 +54,9 @@ def check_for_incomplete_attendances(self):
...
-                if reason:
-                    vals = {'check_out': leave_time.strftime(
-                        DEFAULT_SERVER_DATETIME_FORMAT),
-                        'attendance_reason_ids': [(4, reason.id)]}
-                else:
-                    vals = {'check_out': leave_time.strftime(
-                        DEFAULT_SERVER_DATETIME_FORMAT)}
-                att.write(vals)
+            close = not att.employee_id.no_autoclose
+            if close and max_hours and att.open_worked_hours > max_hours:
```
Comment (AaronHForgeFlow) [inline hr_attendance_autoclose/models/hr_attendance.py]: We can put also the condition in an external method, so it is easier to extend, what do you think?

Comment (jarroyomorales) [reply hr_attendance_autoclose/models/hr_attendance.py]: Done @aheficent !

**Review by AaronHForgeFlow — APPROVED** [review]

File: hr_attendance_autoclose/models/hr_attendance.py (hunk 1 above)
Comment (AaronHForgeFlow) [reply hr_attendance_autoclose/models/hr_attendance.py]: great!

**Review by etobella — APPROVED** [review]
LGTM. It is better to add some hooks sometimes.

**Review by LoisRForgeFlow — APPROVED** [review]
LGTM

**Comment by jarroyomorales** [comment]
@pedrobaeza @jbeficent Can we merge this?

**Comment by AaronHForgeFlow** [comment]
Ready to merge

**Comment by AaronHForgeFlow** [comment]
prstalesmall

**Comment by jarroyomorales** [comment]
@alexey-pelykh Can you merge this please?

**Review by alexey-pelykh — APPROVED** [review]
Code lgtm

**Pull request merged**